languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/pt-BR/grammar.xml

```diff
@@ -887,34 +889,103 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+        <example correction="fizesse">Achei que <marker>fizessem</marker> algumas décadas!</example>
     </rule>
 
+    <rule id="VAO_FAZER_TEMPO" name="Vai fazer tempo" default="temp_off">
+        <antipattern>
+            <token postag_regexp="yes" postag="N..P.+|PP3.P.+"/>
+            <token inflected="yes" postag_regexp="yes" postag="VMI.3P.">ir</token>
+            <token>fazer</token>
+            <example>Os ciclistas vão fazer algumas horas na estrada todo dia.</example>
+            <example>Eles iam fazer 18 anos na mesma semana.</example>
+        </antipattern>
+        <antipattern>
+            <token inflected="yes" postag_regexp="yes" postag="VMI.3P.">ir</token>
+            <token>fazer</token>
+            <token postag_regexp="yes" postag="Z.C.+|D[NI].+"/>
+            <token regexp="yes" postag_regexp="yes" postag="NC.P.+">&expressoes_de_tempo_simples;</token>
+            <token>de</token>
+            <token postag_regexp="yes" postag="NC.+"/>
+            <example>Eles vão fazer trinta anos de carreira mês que vem.</example>
+            <example>Vão fazer 10 anos de idade.</example>
+        </antipattern>
+        <pattern>
+            <marker>
+                <token inflected="yes" postag_regexp="yes" postag="VMI.3P.">ir</token>
+                <token>fazer</token>
+            </marker>
+            <token postag_regexp="yes" postag="R." min="0" max="-1"/>
+            <token postag_regexp="yes" postag="Z.C.+|D[NI].+"/>
+            <token regexp="yes" postag_regexp="yes" postag="NC.P.+">&expressoes_de_tempo_simples;</token>
+        </pattern>
+        <message>O verbo &quot;fazer&quot; no sentido temporal é impessoal. Substitua &quot;\1 \2&quot; por <suggestion><match no="1" postag_regexp="yes" postag="(VMI.3)P(.+)" postag_replace="$1S$2"/> \2</suggestion>.</message>
+        <example correction="Vai fazer"><marker>Vão fazer</marker> muitos anos desde a última vez.</example>
+        <example correction="ia fazer">Em janeiro <marker>iam fazer</marker> seis meses, e ele estivesse vivo...</example>
+        <example correction="vai fazer">Já <marker>vão fazer</marker> 10 minutos que estamos esperando!</example>
+    </rule>
     <!-- FIM VERBO FAZER SENTIDO TEMPORAL É IMPESSOAL-->
-    <!-- ********************************************************** -->
-    <!-- REDUNDÂNCIA EXPORTOU PARA FORA -->
 
+    <rule id="NAO_E_ADJ" name="Não e + adjetivo, falta de acento" default="temp_off">
+        <pattern>
+            <token>não</token>
+            <marker>
+                <token>e</token>
+            </marker>
+            <token postag_regexp="yes" postag="R." min="0" max="-1"/>
+            <token postag_regexp="yes" postag="A.+"/>
+        </pattern>
+        <message>Possível erro de digitação: será que você quis escrever <suggestion>é</suggestion>?</message>
+        <short>Possível erro de digitação.</short>
```

Comment:
Maybe we could have a comma suggestion here as well to deal with possible ambiguous sentences. 

(But we could also wait and see if such FPs will appear in the diff results in the first place, before we get ahead of ourselves)

Reply:
Yes, fair, I'll give myself a to-do for this.